Generate CI images with much DRY-er Python dockerfile generating script

jagerman opened the pull request:
~~Adds cached images that install just the base tools (which are quite substantial), but not the deps (which get installed as part of the build).~~

Replaces the dockerfiles with a Python script that generates them, and adds a bunch of them that are useful for builds debs (namely the new `{debian,ubuntu}-[release]-builder` images, which have only basic deb building tools installed).

jagerman committed:
Add a bunch of base CI images for building debs

Installs just the base tools (which are quite substantial), but not the
deps (which get installed as part of the build).

majestrate commented on contrib/ci/docker/20-deb-builder-bullseye.dockerfile:
```diff
@@ -0,0 +1,10 @@
+ARG ARCH=amd64
+FROM registry.oxen.rocks/lokinet-ci-debian-bullseye-base/${ARCH}
```
what if we added an 
`ARG FLAVOR=bullseye`
and an 
`ARG DISTRO=debian`

and then do just one dockerfile or something?

jagerman replied:
It might be easier to just replace the whole crappy dockerfile BS (such as lack of includes) with a sane language that generates a dockerfile on the fly.

majestrate replied:
heh

jagerman committed:
Replace dockerfiles with a Python script

jagerman committed:
Add --no-cache option; fix manifest generation

jagerman committed:
Parallelize builds

jagerman committed:
Make it snazzy

Shows statuses that "fill in" when run interactively.

jagerman committed:
Fix i386 builds

jagerman committed:
Add hacks; add --distro to rebuild just one

Bionic is broken and needs a hack to create some dumb pgm include
directory.

Add --distro cli argument so that you can rebuild just one distro.

jagerman committed:
Fix race condition in interactive output

jagerman committed:
Install g++-8 on bionic
